lib/rubygems.rb

```diff
@@ -1337,6 +1337,9 @@ def default_gem_load_paths
 
   require 'rubygems/defaults/operating_system'
 rescue LoadError
+  # Ignored
+rescue Exception => e
```

Comment:
In general, it's not recommended to `rescue Exception`, and it's best to let other exceptions go. I say let's do the same here and rescue only `StandardError`.

Reply:
Regarding wording, I would say something like

> Loading the `rubygems/defaults/operating_system.rb` file caused an error. This file is owned by your OS, not by rubygems upstream. Please find out which OS package this file belongs to and follow the guidelines from your OS to report the problem and ask for help.

Reply:
@deivid-rodriguez While I agree with you on best practices, I think it could be better for us to capture any Exception because we are not going to silence the error and it will be raised again with a new message.

The problem with using `StandardError` is it doesn't cover `SyntaxError` which I thought could be important to us since we don't own the `rubygems/defaults/operating_system`. However, I'm happy to change the code to rescue `StandardError` instead.

Reply:
Yeah, I was thinking along those lines too.

In the case of `SyntaxError`, just like with `LoadError`, I think it's really unlikely that an OS is going to ship a ruby file with invalid syntax, so it feels fine to ignore that.

With other errors, likely say `NoMemoryError`, if they happen, we can't really be sure it's the fault of the `rubygems/defaults/operating_system` file, so I'd say it's better to let them be raised unchanged.

That's my reasoning for rescuing only `StandardError`:, it will cover the real life situations that have already happened, but no others that might or might not result in "false positives".